### ISSUE TITLE
Department disable comments enables a cop for the rest a file even if it is excluded with `Include`/`Exclude`

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#13257]
## Description

When disabling an entire department for some lines, a cop that would normally not run on that file because of `Include` restrictions. See https://github.com/rubocop/rubocop-rails/issues/1365 for the original report. cc @pdobb

## Steps to reproduce the problem
```yaml
# .rubocop.yml
AllCops:
  DisabledByDefault: true

Style/HashSyntax:
  Enabled: true
  Include:
    - foo.rb
# Alternatively:
#  Exclude:
#    - bar.rb
```

```rb
# bar.rb
{ :a => :b }

# rubocop:disable Style
{ :a => :b }
# rubocop:enable Style
```

I expect no offenses here since that cop is not supposed to run on this file:

```
$ bundle exec rubocop
Inspecting 2 files
.C

Offenses:

bar.rb:1:3: C: [Correctable] Style/HashSyntax: Use the new Ruby 1.9 hash syntax.
{ :a => :b }
  ^^^^^

2 files inspected, 1 offense detected, 1 offense autocorrectable
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.4.0) [x86_64-linux]
```

**Comment by stephenprater** [comment]
You don't even need to disable the whole department.

```ruby
class LocksController < ApplicationController
  def show
    @ivar = 1
  end

  def update
    1.upto(1) do |r|
      # rubocop:disable Rails:I18nLocaleTexts
      if true
        flash[:notice] = "The universe is a sensible place!"
      end
      # rubocop:enable Rails:I18nLocaleTexts
    end
  end
end
```

This will fire the HelperInstanceVariable cop where you set `@ivar` in the `show` action.  Removing the block or the if statement fixes it.

**Comment by Earlopain** [comment]
Will be fixed in the next version.

@stephenprater in your example you use `:` as the separator, which terminates the rubocop comment. So it is in practise it is a department disable as well.
